### ISSUE TITLE
[cssom-view] Visual viewport's scale factor in iframe

### ISSUE THREAD

**Issue by upsuper** [w3c/csswg-drafts#8434]
When a user pinch-zoom a page, the scale factor can be accessed via `visualViewport.scale`. However, such information is not available within `<iframe>`. `visualViewport.scale` consistently returns `1` among all browsers I tested, and it doesn't seem to be clarified explicitly in the spec. In addition, if the page in `<iframe>` and outside are not in the same origin, accessing to `parent.visualViewport` is restricted.

It would be good to make it possible for pages inside `<iframe>` to have access to such scale factor, as well as allowing to listen to the event that informs its change. It could potentially be done either by
* making `visualViewport.scale` always refer the top level scale factor, or
* making `parent.visualViewport` (or part of it) accessible even across origin.

I don't think enabling accessing this scale factor leaks much information across origins.

For context, we have logic to pick image based on the rendering size of it on the screen. However, when a user embeds our page in a small `<iframe>` within their own page (especially when such page is viewed on a mobile device), we may pick a very small image to show for performance reason (traffic, memory, etc.). When the viewer pinch-zoom to look at the details, we are unable to automatically swap in a larger image because we are not aware of the scale change, and as a result, they may see pixelated image.

**Comment by bokand** [comment]
FWIW - this was an intentional choice, there's some discussion of it in the original [WICG draft spec](https://wicg.github.io/visual-viewport/index.html#the-visualviewport-interface). Conceptually, each frame has its own visual viewport which describes how its document is scaled and presented. It happens that most (all?) browsers don't allow scaling subframes individually but the model in the spec is that a visual viewport is tied to an individual document.

Even seemingly minor leaks across origins can be used as a side channel so I'm not sure this would be safe.

**Comment by upsuper** [comment]
I agree that conceptually it would be good for each frame to have its own visual viewport to indicate how it's scaled and presented, but in practice I'm not sure how useful it is. Parent document often wants the embedded content to look in a certain way, so it may be undesirable if user can change the viewport / zoom independently. Even if the parent document doesn't care, it might cause confusion for users if browser supports doing different zooming in different parts of the document. Given this, and the fact that no browser seemingly supports that, it might be good to revisit the choice.

If the `scale` can be represented in a way that is independent to the document original dimensions, the scale would be purely from the user gesture, and the only thing it can possibly leak is the size of content on the page, which I doubt would be a useful side channel. OTOH, if we care about leaking size of content, this is already leaking content inside iframe to the parent document crossing the origin boundary, I don't see why the reverse would be a larger problem given parent document generally have more control over content inside iframe than the reverse.
